Sources/Fuzzilli/Corpus/MarkovCorpus.swift

```diff
@@ -0,0 +1,150 @@
+                b.callFunction(objectConstructor, withArgs: [])
+                add(b.finalize())
+            }
+        }
+    }
+
+    /// Adds an individual program to the corpus
+    public func add(_ program: Program) {
+        if program.size > 0 {
+            let execution = fuzzer.execute(program)
+            guard execution.outcome == .succeeded else { return }
+            if let aspects = fuzzer.evaluator.evaluate(execution) {
+                add(program, aspects)
+            }
+        }
+    }
+
+    /// Adds multiple programs to the corpus.
+    public func add(_ programs: [Program]) {
+        logger.info("Markov Corpus import of \(programs.count) programs")
+        for (index, prog) in programs.enumerated() {
+            if index % 500 == 0 {
+                logger.info("Markov Corpus import at \(index) of \(programs.count)")
+            }
+            add(prog)
+        }
+    }
+
+    public func add(_ program: Program, _ aspects: ProgramAspects) {
+        if program.size > 0 {
+            self.allIncludedPrograms.append(program)
+            let edges = aspects.toEdges()
+            for e in edges {
+                self.edgeMap[e] = program
+            }
+        }
+    }
+
+    // Switch evenly between programs in the current queue and all programs available to the corpus
+    public func randomElement() -> Program {
+        assert(self.size > 0)
+        if let prog = self.programExecutionQueue.randomElement(), probability(0.5) {
+            assert(prog.size != 0)
+            return prog
+        }
+        let prog = self.allIncludedPrograms.randomElement()!
+        assert(prog.size != 0)
+        return prog
+    }
+
+    public func getNextSeed() -> (seed: Program, energy: UInt64) {
+        self.totalExecs += 1
+        assert(self.size > 0)
+        // Only do computationally expensive work choosing the next program when there is a solid
+        // baseline of execution data
+        if self.totalExecs > 500 {
+            // Check if more programs are needed
+            if self.programExecutionQueue.count == 0 {
+                self.regenProgramList()
+            }
+            assert(self.programExecutionQueue.count > 0)
+            let prog = self.programExecutionQueue.popLast()!
+            return (prog, self.energyBase())
+        } else {
+            let element = self.allIncludedPrograms.randomElement()!
+            return (element, 1)
+        }
+    }
+
+    // Regenerates the internal program list, with dropoff to ensure some diversity among instances
+    // Ends up with ~ 1/32th of the Corpus
+    private func regenProgramList() {
+        assert(self.programExecutionQueue.count == 0)
+        // TODO: Hook things up so that configurable constant "numConsecutiveMutations" is used rather than 5
+        let edges = self.fuzzer.evaluator.smallestEdges(desiredEdgeCount: UInt64(self.size/8), expectedRounds: self.energyBase() * 5)!.toEdges()
```

Comment:
Nit: here and elsewhere, I think you can remove the `self` if not in a closure to stay consistent with the existing code